systemd/systemd.go

```diff
@@ -589,6 +593,33 @@ func (s *systemd) getGlobalUserStatus(unitNames ...string) ([]*UnitStatus, error
 	return sts, nil
 }
 
+func (s *systemd) CurrentMemoryUsage(unit string) (quantity.Size, error) {
+	out, err := s.systemctl("show", "--property", "MemoryCurrent", unit)
+	if err != nil {
+		return 0, osutil.OutputErr(out, err)
+	}
+	// strip the MemoryCurrent= from the output
+	splitVal := strings.SplitN(strings.TrimSpace(string(out)), "=", 2)
+	if len(splitVal) != 2 {
+		return 0, fmt.Errorf("invalid property format from systemd for MemoryCurrent")
+	}
+
+	trimmedVal := strings.TrimSpace(splitVal[1])
+
+	// if the unit is inactive or doesn't exist, the memory usage can be
+	// reported as "[not set]"
+	if trimmedVal == "[not set]" {
+		return 0, fmt.Errorf("memory usage unavailable")
+	}
+
+	intVal, err := strconv.Atoi(trimmedVal)
+	if err != nil {
+		return 0, fmt.Errorf("invalid property format for memory: cannot parse %q as an integer", trimmedVal)
```

Comment:
for consistency:

"invalid property value from systemd for MemoryCurrent: cannot parse %q as an integer"

Reply:
looks good thanks